### PR TITLE
Rdm 8747: supplementary data db changes

### PR DESCRIPTION
**Pull request by Thor-tech-of-metal**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ccd/domain/model/definition/CaseDetails.java [hunk 1]
```diff
@@ -208,6 +215,22 @@ public void setLastStateModifiedDate(LocalDateTime lastStateModifiedDate) {
         this.lastStateModifiedDate = lastStateModifiedDate;
     }
 
+    public Map<String, JsonNode> getSupplementaryData() {
+        return supplementaryData;
+    }
+
+    public void setSupplementaryData(Map<String, JsonNode> supplementaryData) {
+        this.supplementaryData = supplementaryData;
+    }
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/domain/model/definition/CaseDetails.java:224]: we don't want to allow getting and setting the whole supplementary data. Consider providing methods to instead get and set specific properties

File: src/main/resources/db/changelog/db.changelog-master.xml [hunk 2]
```diff
@@ -32,4 +32,5 @@
     <include file="db/changelog/db.changelog-RDM-6111.xml"/>
     <include file="db/changelog/db.changelog-RDM-4641.xml"/>
     <include file="db/changelog/db.changelog-RDM-8485.xml"/>
+    <include file="db/changelog/db.changelog-RDM-8747.xml"/>
```
Comment (mario-paniccia) [inline src/main/resources/db/changelog/db.changelog-master.xml:35]: can you push the file too please

File: src/main/java/uk/gov/hmcts/ccd/domain/model/definition/CaseDetails.java (hunk 1 above)
Comment (Thor-tech-of-metal) [reply src/main/java/uk/gov/hmcts/ccd/domain/model/definition/CaseDetails.java:224]: it is work in progress I added for my own benefit while I am doing the work . But understood

File: src/main/java/uk/gov/hmcts/ccd/data/casedetails/CaseDetailsEntity.java [hunk 3]
```diff
@@ -1,31 +1,43 @@
 package uk.gov.hmcts.ccd.data.casedetails;
```
Comment (Thor-tech-of-metal) [inline src/main/java/uk/gov/hmcts/ccd/data/casedetails/CaseDetailsEntity.java:1]: src/main/java/uk/gov/hmcts/ccd/data/casedetails/CaseDetailsEntity.java will be removed from the PR

File: src/main/java/uk/gov/hmcts/ccd/domain/model/definition/CaseDetails.java [hunk 4]
```diff
@@ -30,7 +30,8 @@
 import static uk.gov.hmcts.ccd.domain.model.definition.FieldTypeDefinition.CASE_PAYMENT_HISTORY_VIEWER;
```
Comment (Thor-tech-of-metal) [inline src/main/java/uk/gov/hmcts/ccd/domain/model/definition/CaseDetails.java:30]: src/main/java/uk/gov/hmcts/ccd/domain/model/definition/CaseDetails.java  will be removed from the PR

File: src/main/resources/db/changelog/db.changelog-RDM-8747.xml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+    <changeSet id="rdm-8747" author="tobias.derose@hmcts.net">
+        <addColumn tableName="case_data">
+            <column name="supplementary_data" type="jsonb">
+                <constraints nullable="true"/>
+            </column>
+            <column name="supplementary_data_last_modified" type="timestamp">
+                <constraints />
```
Comment (mario-paniccia) [inline src/main/resources/db/changelog/db.changelog-RDM-8747.xml:11]: we have this as initially null?  and then will be gradually populated whenever supplementary data is written? Would this cause problems to the LS sql query? let's verify

Comment (Thor-tech-of-metal) [reply src/main/resources/db/changelog/db.changelog-RDM-8747.xml:11]: Yes it has been tested .  When cases are created  supplementary_data_last_modified is null. LogStash keep running and execution queries without error.

Comment (mario-paniccia) [reply src/main/resources/db/changelog/db.changelog-RDM-8747.xml:11]: ok cool, having it as initially null is fine then.

**Pull request merged**